core/bindings.rs

```diff
@@ -271,6 +271,54 @@ pub extern "C" fn host_initialize_import_meta_object_callback(
   let main_key = v8::String::new(scope, "main").unwrap();
   let main_val = v8::Boolean::new(scope, info.main);
   meta.create_data_property(scope, main_key.into(), main_val.into());
+
+  let resolve_key = v8::String::new(scope, "resolve").unwrap();
+  let val = v8::Function::new(scope, import_meta_resolve).unwrap();
+  val.set_name(resolve_key);
+  meta.set(scope, resolve_key.into(), val.into());
+}
+
+fn import_meta_resolve(
+  scope: &mut v8::HandleScope,
+  args: v8::FunctionCallbackArguments,
+  mut rv: v8::ReturnValue,
+) {
+  if args.length() > 1 {
+    return throw_type_error(scope, "Invalid arguments");
+  }
+
+  let tc_scope = &mut v8::TryCatch::new(scope);
+  let maybe_arg_str = args.get(0).to_string(tc_scope);
+  if maybe_arg_str.is_none() {
+    return throw_type_error(tc_scope, "Invalid arguments");
+  }
+  let specifier = maybe_arg_str.unwrap();
+  let referrer = {
+    let receiver = args.this();
+    let url_key = v8::String::new(tc_scope, "url").unwrap();
+    let url_prop = receiver.get(tc_scope, url_key.into()).unwrap();
+    url_prop.to_rust_string_lossy(tc_scope)
+  };
```

Comment:
Good point, I'll rework it

Reply:
Addressed, using an own property

Reply:
Actually this approach doesn't work - there are WPT test cases that check for all properties on `import.meta` to be writable/enumerable/configurable, so I can't store field for internal this way. I also cannot use "internal fields" API from V8, because we don't control creation of the object. I'll try to use "private property" API, but it's unstable currently.